internal/db/schema/state.go

```diff
@@ -0,0 +1,57 @@
+package schema
+
+import (
+	"context"
+)
+
+const nilVersion = -1
+
+type migrationState struct {
+	// devMigration is true if the database schema that would be applied by
+	// InitStore would be from files in the /dev directory which indicates it would
+	// not be safe to run in a non dev environment.
+	devMigration bool
+
+	// binarySchemaVersion provides the database schema version supported by
+	// this binary.
+	binarySchemaVersion int
+
+	upMigrations   map[int][]byte
+	downMigrations map[int][]byte
+}
+
+var migrationStates = make(map[string]migrationState)
```

Comment:
a comment here about how this is initialized via the generator might be nice.

Reply:
yeah or neah on the comment?

Reply:
Added comments. PTAL.